### PR TITLE
Workflow for automating the release

### PR DESCRIPTION
**Pull request by tim-vd-aardweg**

### CODE REVIEW

**Review by MRVermeulenDeltares — COMMENTED** [review]
@tim-vd-aardweg lets discuss these points

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1 +1,77 @@
...
+    steps:
+      - name: Check out the code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
```
[the remark this answers is not in the document]
Comment (tim-vd-aardweg) [reply .github/workflows/release.yml:31]: Let's make it 3.8

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1 +1,77 @@
...
+        description: "The new minor version."
+        required: true
+        type: string
+      patch:
+        description: "The new patch version."
+        required: true
+        type: string
+
+jobs:
+  prepare-release:
```
[the remark this answers is not in the document]
Comment (tim-vd-aardweg) [reply .github/workflows/release.yml:20]: Let's change the name!

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -1 +1,77 @@
...
+
+      - name: Publish package to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          password: ${{ secrets.PYPI_API_TOKEN }}
+
+      - name: Create GitHub release
+        uses: softprops/action-gh-release@v1
+        with:
+          name: Release v${{ major }}.${{ minor }}.${{ patch }}
```
[the remark this answers is not in the document]
Comment (tim-vd-aardweg) [reply .github/workflows/release.yml:74]: Yes! This is correct accordingto the release steps in https://deltares.github.io/HYDROLIB-core/0.4.1/guides/contributing/#making-a-release-on-github-and-pypi

File: .github/workflows/release.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tim-vd-aardweg) [reply .github/workflows/release.yml:31]: Done!

File: .github/workflows/release.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tim-vd-aardweg) [reply .github/workflows/release.yml:20]: Done!

**Review by arthurvd — APPROVED** [review]
Looks good to me (@tim-vd-aardweg already fixed a wrong full uppercase CHANGELOG.md).
In the future we could consider adding a "draft" option that upload to the test PyPI, and creates a draft release, but for now let's do a trial run for the 0.5.0 directly.

**Pull request merged**